astropy/nddata/nduncertainty.py

```diff
@@ -933,3 +981,10 @@ def _propagate_divide(self, other_uncert, result_data, correlation):
 
     def _data_unit_to_uncertainty_unit(self, value):
         return 1 / value ** 2
+
+    def _convert_to_variance(self):
+        return VarianceUncertainty(self.array ** -1, unit=self.unit ** -1)
+
+    @classmethod
+    def _convert_from_variance(cls, var_uncert):
+        return cls(var_uncert.array ** -1, unit=var_uncert.unit ** -1)
```

Comment:
Same issue as https://github.com/astropy/astropy/pull/12057/files#r854225811 when `self.unit is None`